### PR TITLE
[CDEC-334] Create delegation-test package and move Arbitrary instances

### PR DESCRIPTION
**Pull request by intricate**
## Description

As it stands, the `Arbitrary` instances for data types in `delegation` all exist in `delegation` when they should actually be exposed from a separate package, `delegation-test`. This PR creates this new package and moves over the `Arbitrary` instances.

## Linked issue

[https://iohk.myjetbrains.com/youtrack/issue/CDEC-334](https://iohk.myjetbrains.com/youtrack/issue/CDEC-334)
## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [x] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: delegation/test/cardano-sl-delegation-test.cabal [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name:                cardano-sl-delegation-test
+version:             1.1.0
```
Comment (akegalj) [inline delegation/test/cardano-sl-delegation-test.cabal]: aren't we at 1.2.0 ?

File: pkgs/default.nix [hunk 2]
```diff
@@ -16033,6 +16037,45 @@ doHaddock = false;
...
+, generic-arbitrary
+, QuickCheck
+, stdenv
+, universum
+, unordered-containers
+}:
+mkDerivation {
+
+pname = "cardano-sl-delegation-test";
+version = "1.1.0";
```
Comment (akegalj) [inline pkgs/default.nix]: aren't we at 1.2.0 ?

File: delegation/test/cardano-sl-delegation-test.cabal (hunk 1 above)
Comment (intricate) [reply delegation/test/cardano-sl-delegation-test.cabal]: Supposed to change to 1.3.0. Will get that done.

Comment (intricate) [reply delegation/test/cardano-sl-delegation-test.cabal]: Looks like I changed it but didn't push the commit... my bad.

**Pull request merged**